homeassistant/components/scene/__init__.py

```diff
@@ -6,28 +6,55 @@
 """
 import asyncio
 import logging
-from collections import namedtuple
 
 import voluptuous as vol
 
 from homeassistant.const import (
-    ATTR_ENTITY_ID, SERVICE_TURN_ON, CONF_PLATFORM)
-from homeassistant.helpers import extract_domain_configs
+    ATTR_ENTITY_ID, CONF_PLATFORM, SERVICE_TURN_ON)
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity import Entity
 from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.state import HASS_DOMAIN
+from homeassistant.loader import get_platform
 
 DOMAIN = 'scene'
 STATE = 'scening'
+STATES = 'states'
 
-CONF_ENTITIES = "entities"
+
+def _hass_domain_validator(config):
+    """Validate platform in config for homeassistant domain."""
+    if CONF_PLATFORM not in config:
+        config = {
+            CONF_PLATFORM: HASS_DOMAIN, STATES: config}
+
+    return config
+
+
+def _platform_validator(config):
+    """Validate it is a valid  platform."""
+    p_name = config[CONF_PLATFORM]
+    platform = get_platform(DOMAIN, p_name)
+
+    if not hasattr(platform, 'PLATFORM_SCHEMA'):
+        return config
+
+    return getattr(platform, 'PLATFORM_SCHEMA')(config)
+
+
+PLATFORM_SCHEMA = vol.Schema(
+    vol.All(
+        _hass_domain_validator,
+        vol.Schema({
+            vol.Required(CONF_PLATFORM): cv.platform_validator(DOMAIN)
+        }, extra=vol.ALLOW_EXTRA),
+        _platform_validator
```

Comment:
Isn't this automatically done by `async_setup_component` when it detects a `PLATFORM_SCHEMA` ? 

https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/config.py#L540-L572

Edit: oh realized now that it is not happening for hass platforms because we support the old config too.